src/Geometry/CartesianDiscreteModels.jl

```diff
@@ -16,7 +16,11 @@ struct CartesianDiscreteModel{D,T,F} <: DiscreteModel{D,D}
   function CartesianDiscreteModel(desc::CartesianDescriptor{D,T,F}) where {D,T,F}
     grid = CartesianGrid(desc)
     _grid = UnstructuredGrid(grid)
-    topo = UnstructuredGridTopology(_grid)
+    if any(desc.isperiodic)
+      topo = UnstructuredGridTopology(_grid, desc.isperiodic, desc.partition)
```

Comment:
Do not name this function `UnstructuredGridTopology` since it is not a general constructor of `UnstructuredGridTopology` objects. 

Name the function e.g.  `_cartesian_grid_topology_with_periodic_bcs`

Reply:
Shall I move it from UnstructuredGrids.jl to CartesianGrids.jl as well?

Reply:
Yes please!